code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -869,3 +869,59 @@
 		log_game("DYNAMIC: Not enough players volunteered for the ruleset [name] - [candidates.len] out of [required_candidates].")
 		return FALSE
 	return TRUE
+
+//////////////////////////////////////////////
+//                                          //
+//           NINJA      (GHOST)             //
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/midround/from_ghosts/ninja
+	name = "Space Ninja"
+	midround_ruleset_style = MIDROUND_RULESET_STYLE_HEAVY
+	antag_flag = ROLE_NINJA
+	required_type = /mob/dead/observer
+	antag_datum = /datum/antagonist/ninja
+	enemy_roles = list(JOB_NAME_SECURITYOFFICER, JOB_NAME_DETECTIVE, JOB_NAME_WARDEN, JOB_NAME_HEADOFSECURITY, JOB_NAME_CAPTAIN)
+	required_enemies = list(2,2,2,2,2,2,2,2,2,2)
```

Comment:
I have been trying out lower sec requirements, around `list(2,2,1,1,1,1,1,1,0,0)` or so. but with ninja I honestly don't think it should ever spawn nosec so that's probably good

Reply:
Yea, I agree. This whole system seems a little strange to me since at higher threats you need more security to deal with the increased number of threats, however at higher threats you don't need any security for some threats. Space ninja specifically requires sec to deal with them since they aren't an antag that the crew can reasonably fight without getting bwoinked unlike most others.